src/util/system.h

```diff
@@ -327,7 +333,7 @@ class ArgsManager
      * @param nDefault (e.g. 1)
      * @return command-line argument (0 if invalid number) or default value
      */
-    int64_t GetArg(const std::string& strArg, int64_t nDefault) const;
+    int64_t GetIntArg(const std::string& strArg, int64_t nDefault) const;
```

Comment:
I think I finally figured out how you can go a little bit further than the above, ending up with something like:

```c++
struct NetSettings
{
    int64_t blockreconstructionextratxn;
    int64_t maxorphantx;
    bool capturemessages;

    template<typename C, typename... Args>
    static inline void F(Args&... args) {
        return C::Do(args...,
            C::Defn( &NetSettings::blockreconstructionextratxn, "-blockreconstructionextratxn=<n>", strprintf("Extra transactions to keep in memory for compact block reconstructions (default: %u)", DEFAULT_BLOCK_RECONSTRUCTION_EXTRA_TXN), ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS, DEFAULT_BLOCK_RECONSTRUCTION_EXTRA_TXN ),
            C::Defn( &NetSettings::maxorphantx, "-maxorphantx=<n>", strprintf("Keep at most <n> unconnectable transactions in memory (default: %u)", DEFAULT_MAX_ORPHAN_TRANSACTIONS), ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS, DEFAULT_MAX_ORPHAN_TRANSACTIONS ),
            C::Defn( &NetSettings::capturemessages, "-capturemessages", "Capture all P2P messages to disk", ArgsManager::ALLOW_ANY | ArgsManager::DEBUG_ONLY, OptionsCategory::DEBUG_TEST, false )
        );
    }
};

void RegisterNetProcessingArgs(ArgsManager& args)
{
    SettingsRegister<NetSettings>::Register(args);
}

NetSettings GetNetSettings(const ArgsManager& args)
{
    return SettingsRegister<NetSettings>::Get(args);
}

class PeerManagerImpl
{
private:
    const NetSettings m_settings;
    PeerManagerImpl(..., const ArgsManager& args) : m_settings{GetNetSettings(args)), ... { ... }
    ...
};
```

The idea being that this way:
 * it can infer the argument type directly from the type of the struct member so that you can't accidentally specify different types between `args.AddArg<int>` and `Get<bool>`
  * that the settings are const at runtime so can be accessed without any additional locks
  * you only have to access the ArgsManager (and do string parsing) at setup time
  * you don't have to make up lots of new names for everything or add too much boilerplate

Branch built on top of this PR that has the above working at https://github.com/ajtowns/bitcoin/tree/202109-settings_struct